newm-chain/src/main/kotlin/io/newm/chain/grpc/NewmChainService.kt

```diff
@@ -507,4 +508,53 @@ class NewmChainService : NewmChainGrpcKt.NewmChainCoroutineImplBase() {
             throw e
         }
     }
+
+    private val mutexMap = mutableMapOf<String, Mutex>()
+    private val mutexExpiryJobMap = mutableMapOf<String, Job>()
+
+    override suspend fun acquireMutex(request: AcquireMutexRequest): MutexResponse {
+        val mutex = mutexMap.getOrPut(request.mutexName) { Mutex() }
```

Comment:
Since `acquireMutex` can be called from multiple thread, I think it's safer to make it synchronized:
```
 private val mutexMap = Collections.synchronizedMap(mutableMapOf<String, Mutex>())
```

Reply:
`getOrPut()` is already concurrency safe. 
![image](https://github.com/projectNEWM/newm-server/assets/245918/62f1dd03-4d01-4d35-8136-5691432ec898)

Reply:
I've updated to `ConcurrentHashMap` to further ensure safety here.

Reply:
Yeah,  change to `ConcurrentHashMap` and use `getOrDefault()`.